data/binding/queue.go

```diff
@@ -1,30 +1,33 @@
 package binding
 
 import (
-	"sync"
+	"runtime"
 
-	"fyne.io/fyne/v2/internal/async"
+	"fyne.io/fyne/v2"
 )
 
-var (
-	once  sync.Once
-	queue *async.UnboundedFuncChan
-)
+var mainGoroutineID uint64
 
-func queueItem(f func()) {
-	once.Do(func() {
-		queue = async.NewUnboundedFuncChan()
-		go func() {
-			for f := range queue.Out() {
-				f()
-			}
-		}()
-	})
-	queue.In() <- f
+func init() {
+	runtime.LockOSThread()
+	mainGoroutineID = goroutineID()
 }
 
-func waitForItems() {
-	done := make(chan struct{})
-	queue.In() <- func() { close(done) }
-	<-done
+func goroutineID() (id uint64) {
+	var buf [30]byte
+	runtime.Stack(buf[:], false)
+	for i := 10; buf[i] != ' '; i++ {
```

Comment:
Is this basically a hash code? This makes me *slightly* nervous about hash collisions. If the [30]byte we get out of runtime.Stack is already unique maybe we just make `type goroutineID [30]byte` (or [20]byte since it seems we only care about buf[10:]) and use `slices.Equal`?

Reply:
It would be good to reuse the other function instead of copying it here.